craft_application/models/project.py

```diff
@@ -33,6 +38,15 @@
 )
 
 
+@dataclasses.dataclass
+class BuildInfo:
+    """Platform build information."""
+
+    build_on: str
```

Comment:
Are `build_on` and `build_for` application-dependent (other than the potential of some apps not supporting all architectures)? If not, maybe `util.platforms` should have `ARCH_TRANSLATIONS_DEB_TO_PLATFORM` be an enum of all supported architectures and make these be those instances?

(A side benefit there would be that we get the 2-way deb <-> platform mapping for free and can just have a `Dict[str, Arch]` to map any additional platform names to the debian architecture)

Reply:
@mr-cal implemented the original architecture mapping and there were problems in doing it 2-way with a single mapping because it's not symmetrical. If we improve this handling we may want to use the same strategy in Snapcraft.

Reply:
I think you are referring to [this](https://github.com/snapcore/snapcraft/pull/3821/files#diff-7aadd8b7de9c97da932b8546600568e5ebeafffddfabcf4bd15023bd5c1da925R49-R50).

That was done as part of a mega-PR and I suspect spending more time creating an enum or dataclass would solve the problem better than my two dictionaries.

Reply:
We should work on a unified implementation of architecture mapping (maybe including os/release querying too?) that could be used by libs and apps. I propose we create a new issue and keep the focus of this one on the build plan handling.